.github/workflows/python-publish.yml

```diff
@@ -0,0 +1,45 @@
+# This workflow will upload a Python Package using Twine when a release is created
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python#publishing-to-package-registries
+
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+name: Upload Python Package
+
+on:
+  push:
+    branches:
+      - "*"
+
+permissions:
+  contents: read
+
+jobs:
+  deploy:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python
+      uses: actions/setup-python@v3
+      with:
+        python-version: '3.7'
+    - name: Install dependencies
+      run: |
+        python setup.py sdist
+    - name: Publish distribution to PyPI
+      if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags')
```

Comment:
Cool just to confirm this means it will only ever run this on CI pipelines run from tags right?

Reply:
I pulled it from https://github.com/pypa/gh-action-pypi-publish#usage

```
A common use case is to upload packages only on a tagged commit, to do so add a filter to the step:

  if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags')

```

In my limited testing it appears to do the trick